UN-OCHA/unocha-site
# [UNO-649] Event date

Refs: UNO-649

This PR changes the type of the field_event_date to a daterange with timezone and provides a template with some logic to display the dates and time (and timezone) separately.

It also adds 2 theme suggestions for fields with the parent entity view mode.

### Templates

Currently there is only a template for the event date field in "full" view mode. That can be copied for other view modes or the `full` view mode part can be removed from the template name to apply to all view modes.

Ex: `field--node--event--full--field-event-date.html.twig` --> `field--node--event--field-event-date.html.twig`

### Logic

The template contains the logic to display the start, end and time. Notably it does the following:

- If the `end` date is the same as the `start` date, then only the start date is shown. Ex: `15 June 2023`.
- if the `start` date has the same year and month as the `end` date then only the day is shown for the start date. Ex: `14-15 June 2023`.
- if the `start` date has the same year but different month, then the day and month are displayed. Ex: `14 May-15 June 2023`.
- Otherwise the day month and year is displayed. Ex: `14 May 2022-15 June 2023`.

### Deployment

**Warning:** please make sure to backup your DB and restore it before checking out another branch. No need though, after this PR is merged.

There is a update hook to perform the conversion of the field so it's important to run the following in order:

1. Checkout the branch
2. Recreate the image
3. Recreate the container
4. Run `drush cr`
5. Run `drush updatedb --no-post-updates` --> that will convert the field
6. Run `drush cim`
7. Run `drush updatedb`

Nothing special to deploy to `dev` normally. The job will basically do the above.

### Editorial notes

The date and time entered in the form need to correspond to the date and time in the selected timezone. That means that if the event date is `14 June 2023` at `9:00 AM EDT` then enter `06/14/2023` and `09:00:00 AM` in the start date and time and select `America/New York` as timezone.

**form**

<img width="864" alt="Screenshot 2023-06-02 at 14 46 07" src="https://github.com/UN-OCHA/unocha-site/assets/696348/36d72d08-1e2b-416b-9be9-5736d3af8ec5">

**result**

<img width="335" alt="Screenshot 2023-06-02 at 14 45 59" src="https://github.com/UN-OCHA/unocha-site/assets/696348/ced91526-ae80-464f-821d-3e6957c3042f">

### Tests

1. Follow the deployment steps
2. Edit or create an event set the dates as per the editorial notes above
3. Save
4. Check that the dates and time are displayed as per the logic described above.
